### PR TITLE
Add FlySafe live tracking

### PR DESCRIPTION
**Pull request by Lenart12**
This PR adds live tracking for [FlySafe](https://flysafe.pro), a paragliding social network popular in 🇸🇮! 

I am not affiliated with FlySafe, but I do use it. I have contacted the owner and he is aware of this integration (he even helped me :D)

Tested working in sim and on Garmin Venu 3
image

### CODE REVIEW

File: source/MySettings.mc [hunk 1]
```diff
@@ -585,6 +585,7 @@ class MySettings {
     return LangUtils.readKeyNumber(App.Properties.getValue("userLivetrack24Frequency"), 0);
   }
   function saveLivetrack24Frequency(_iValue as Number) as Void { 
+    setLivetrack24Frequency(_iValue);
```
Comment (ydutertre) [inline source/MySettings.mc]: Can you describe the bug? The setLiveTrack24Frequency() function (and all the other set funcctions) is called when the user exits the settings screen via the MyApp.onSettingsChanged() function, thus updating the iLivetrack24FrequencySeconds variable.
So the out of sync of the iLivetrack24FrequencySeconds variable only occurs between the moment the user sets the new value, and the moment they exit the settings (which they need to do to use the app, so in practice it's never out of sync).

I'd like to remove this line (and the one for STL, and the one you also added for FS), unless you tell me I'm missing something!

Comment (Lenart12) [reply source/MySettings.mc]: You are correct then, while I was developing I noticed that It didnt change instantly when exiting the picker, but I didnt try exiting the settings. I will remove the commit.

**Comment by Lenart12** [comment]
I will have to change which altitude its reporting. Gps one is wildly inaccurate... :D

File: source/MyFlySafeLivetrack.mc [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+      "data" => "[{" +
+        Lang.format("\"id\": \"$1$\", ", [xAuthToken])  +   // Auth token
+        Lang.format("\"lat\": $1$, ", [locDeg[0]])      +   // Latitude in degrees
+        Lang.format("\"lon\": $1$, ", [locDeg[1]])      +   // Longitude in degrees
+        Lang.format("\"accuracy\": $1$, ", [iAccuracy]) +   // Accuracy in meters
+        Lang.format("\"alt\": $1$, ", [_fAltitude])     +   // Altitude in meters
+        Lang.format("\"speed\": $1$, ", [_info.speed])  +   // Speed in m/s
+        Lang.format("\"heading\": $1$, ", [headingDeg]) +   // Heading in degrees
+        Lang.format("\"time\": $1$000, ", [_info.when.value()]) + // Unix time in milliseconds (3 zeros added to the end),
+                                                                  // NOTE: This will break in 2038, but this is a Garmin problem for the whole SDK... LOL!
```
Comment (ydutertre) [inline source/MyFlySafeLivetrack.mc:110]: love this

File: source/MyProcessing.mc [hunk 3]
```diff
@@ -433,6 +433,14 @@ class MyProcessing {
           $.oMySportsTrackLive.iCounter++;
         }
         
+        //FlySafe
+        if ($.oMyActivity != null && $.oMySettings.iFlySafeLivetrackFrequencySeconds != 0) {
+          if ($.oMyFlySafeLivetrack.iCounter % $.oMySettings.iFlySafeLivetrackFrequencySeconds == 0) {
+            $.oMyFlySafeLivetrack.updatePosition(_oInfo, oLocation, self.fAltitude);
```
Comment (ydutertre) [inline source/MyProcessing.mc:439]: Just to make sure I understand, there's no concept of starting and stopping a session/flight in FlySafe? Just sending position info is enough?

**Comment by ydutertre** [comment]
Looks good to me, I'm merging this and will release a new version soon. Very curious about how FS works without an explicit session/flight, just throwing positions at it!

File: source/MyProcessing.mc (hunk 3 above)
Comment (Lenart12) [reply source/MyProcessing.mc:439]: This is correct, no session required. I was surprised as well but it made the whole thing a lot easier so I'm not complaining :D

**Comment by ydutertre** [comment]
App has been compiled and update submitted. Thanks so much for this!
Updated version (v3.50) should be available on Garmin store in a few hours (the Garmin store is weird, it will immediately publish my app updates one time out of two, and the next time it takes several hours. Then immediate at next update. Then several hours. Etc. It's systematic)

Edit: actually it got published immediately, maybe Garmin fixed the behavior above :)

**Pull request merged**